### PR TITLE
WebUI integration tests

### LINKED ISSUE #16: Setup WebUI (MVC) integration tests
All the controllers and each of the actions in each of the controllers should go through an integration test to verify that they are working properly.

Requires:
- A factory to create a server for each test (ASP Core Server) based on the needs of the test
- Since we need to test StatusCodes we should send Http Requests to this newly created server to verify.

### PR DESCRIPTION
**Pull request by misha130**
Closes https://github.com/codidact/core/issues/16

### CODE REVIEW

File: tests/WebUI.IntegrationTests/CustomWebApplicationFactory.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+using System;
+using System.Linq;
+
+namespace Codidact.WebUI.IntegrationTests
+{
+    /// <summary>
+    /// This factory creates an .net Core server with the configuration provided
+    /// for the purpose of testing against it
+    /// </summary>
+    public class CustomWebApplicationFactory<TStartup> : WebApplicationFactory<Startup> where TStartup : class
```
Comment (asynts) [inline tests/WebUI.IntegrationTests/CustomWebApplicationFactory.cs]:
~~~suggestion
    public class CustomWebApplicationFactory<TStartup> : WebApplicationFactory<TStartup> where TStartup : class
~~~
typo?

Comment (misha130) [reply tests/WebUI.IntegrationTests/CustomWebApplicationFactory.cs]: Typo, yes. Nice catch. Thanks

File: tests/WebUI.IntegrationTests/IntegrationTestHelper.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        public static StringContent GetRequestContent(object obj)
+        {
+            return new StringContent(JsonConvert.SerializeObject(obj), Encoding.UTF8, "application/json");
+        }
+
+        public static async Task<T> GetResponseContent<T>(HttpResponseMessage response)
+        {
+            var stringResponse = await response.Content.ReadAsStringAsync();
+
+            var result = JsonConvert.DeserializeObject<T>(stringResponse);
```
Comment (asynts) [inline tests/WebUI.IntegrationTests/IntegrationTestHelper.cs]:
~~~diff
--- a/tests/WebUI.IntegrationTests/IntegrationTestHelper.cs
+++ b/tests/WebUI.IntegrationTests/IntegrationTestHelper.cs
@@ -16,9 +16,7 @@ namespace Codidact.WebUI.IntegrationTests
         {
             var stringResponse = await response.Content.ReadAsStringAsync();
 
-            var result = JsonConvert.DeserializeObject<T>(stringResponse);
-
-            return result;
+            return JsonConvert.DeserializeObject<T>(stringResponse);
         }
     }
 }
~~~

Comment (misha130) [reply tests/WebUI.IntegrationTests/IntegrationTestHelper.cs]: No, this is fine. Its easier to debug if its split in to two lines.

Comment (ArtOfCode-) [reply tests/WebUI.IntegrationTests/IntegrationTestHelper.cs]: ^ as asynts says, this can be a straight `return` without assignment.

**Review by ArtOfCode- — CHANGES_REQUESTED** [review]

File: tests/WebUI.IntegrationTests/IntegrationTestHelper.cs (hunk 2 above)
Comment (misha130) [reply tests/WebUI.IntegrationTests/IntegrationTestHelper.cs]: Well, majority rule. Alright, will do

**Review by ArtOfCode- — APPROVED** [review]

**Pull request merged**